hibernate-core/src/test/java/org/hibernate/orm/test/query/hql/AggregateFilterClauseTest.java

```diff
@@ -0,0 +1,234 @@
+		scope.inTransaction(
+				session -> {
+					Double expected = 13D;
+					Query q = session.createQuery( "select max(eob.theDouble) from EntityOfBasics eob" );
+					assertEquals( q.getSingleResult(), expected, "expected " + expected + ", got " + q.getSingleResult() );
+				}
+		);
+	}
+
+	@Test
+	public void testMaxWithFilterClause(SessionFactoryScope scope) {
+		scope.inTransaction(
+				session -> {
+					Double expected = 6D;
+					Query q = session.createQuery( "select max(eob.theDouble) filter(where eob.theBoolean = true) from EntityOfBasics eob" );
+					assertEquals( q.getSingleResult(), expected, "expected " + expected + ", got " + q.getSingleResult() );
+				}
+		);
+	}
+
+	@Test
+	public void testSimpleCount(SessionFactoryScope scope) {
+		scope.inTransaction(
+				session -> {
+					Long expected = 5L;
+					Query q = session.createQuery( "select count(*) from EntityOfBasics eob" );
+					assertEquals( q.getSingleResult(), expected, "expected " + expected + ", got " + q.getSingleResult() );
+
+					expected = 3L;
+					q = session.createQuery( "select count(eob.theDate) from EntityOfBasics eob" );
+					assertEquals( q.getSingleResult(), expected, "expected " + expected + ", got " + q.getSingleResult() );
+				}
+		);
+	}
+
+	@Test
+	public void testCountWithFilterClause(SessionFactoryScope scope) {
+		scope.inTransaction(
+				session -> {
+					Long expected = 3L;
+					Query q = session.createQuery( "select count(*) filter(where eob.theBoolean = false) from EntityOfBasics eob" );
+					assertEquals( q.getSingleResult(), expected, "expected " + expected + ", got " + q.getSingleResult() );
+
+					expected = 2L;
+					q = session.createQuery( "select count(eob.theDate) filter(where eob.theBoolean = false) from EntityOfBasics eob" );
+					assertEquals( q.getSingleResult(), expected, "expected " + expected + ", got " + q.getSingleResult() );
+				}
+		);
+	}
+
+	@Test
+	// filter clause not allowed here, so should throw an exception
+	public void testAllEveryNoLimitClause(SessionFactoryScope scope) {
+		scope.inTransaction(
+				session -> {
+					// check normal use still works
+					Query q1 = session.createQuery( "select any(eob.id > 0), every(eob.id > 0) from EntityOfBasics eob" );
+					q1.getResultList();
+					q1 = session.createQuery( "select some(eob.id > 0), all(eob.id > 0) from EntityOfBasics eob" );
+					q1.getResultList();
+
+					assertThrows(
+							IllegalArgumentException.class,
+							() -> {
+								Query q2 = session.createQuery( "select every(eob.id > 0) filter(where eob.theBoolean = true) from EntityOfBasics eob" );
+							},
+							"The every predicate doesn't support the filter clause"
+					);
+					assertThrows(
+							IllegalArgumentException.class,
+							() -> {
+								Query q2 = session.createQuery( "select all(eob.id > 0) filter(where eob.theBoolean = true) from EntityOfBasics eob" );
+							},
+							"The all predicate doesn't support the filter clause"
+					);
```

Comment:
Actually, the versions that accept predicates instead of subqueries should support the filter clause, as these functions are just special boolean aggregates. Sorry for the confusion before, but I didn't look carefully enough. The negative test can stay though for when a subquery is used as argument.

Reply:
Just to make sure; are you referring only to the every/all + any/some functions, or are there any more aside those?
Could you clarify what exactly you have in mind for the negative test?

Reply:
If a predicate is passed as argument to these functions, the filter clause should be allowed i.e. the specific test case you have here, should not fail but actually succeed. On the other hand, an expression like `all(select 1) filter (where ..)` should fail though, because that kind of expression is actually only relevant for quantified predicate comparison where the filter clause is illegal.

Reply:
I finally took this 'negative' test out, because I couldn't get any trial based on your last comment to work; if you want to have a go at it, please do, you prob. have a better idea of what you had in mind.